xbmc/cores/dvdplayer/DVDPlayer.cpp

```diff
@@ -835,7 +835,7 @@ bool CDVDPlayer::OpenDemuxStream()
   int64_t len = m_pInputStream->GetLength();
   int64_t tim = m_pDemuxer->GetStreamLength();
   if(len > 0 && tim > 0)
-    m_pInputStream->SetReadRate((unsigned int) (g_advancedSettings.m_readBufferFactor * len * 1000 / tim));
```

Comment:
I think I don't fully understand it. With the above removed code, we told the inputstream to read with the speed of avg_rate * readbufferfactor. I don't see how the max_rate you reference gets set at all now.